src/main/java/racingcar/domain/Car.java

```diff
@@ -1,41 +1,39 @@
 package racingcar.domain;
 
-public class Car {
-    private int distance;
+import racingcar.random.RandomNumberGenerator;
 
-    private static final int DEFAULT_DISTANCE = 0;
-    private static final int MIN_RANDOM_VALUE = 0;
-    private static final int MAX_RANDOM_VALUE = 9;
+public class Car {
     private static final int FORWARD_THRESHOLD = 4;
-    private static final int MOVE_DISTANCE = 1;
 
-    public Car() {
-        this.distance = DEFAULT_DISTANCE;
+    private final CarName name;
+    private final CarDistance distance;
+
+    public Car(String name) {
+        this.name = new CarName(name);
+        this.distance = new CarDistance();
     }
 
-    public void move(int randomValue) {
-        validateRandomValue(randomValue);
+    public void move(RandomNumberGenerator generator) {
+        RandomNumber randomNumber = generator.generate();
 
-        if (shouldMoveForward(randomValue)) {
-            this.distance += MOVE_DISTANCE;
+        if (shouldMoveForward(randomNumber)) {
```

Comment:
```suggestion
        if (randomNumber.movable()) {
```
위와 같이 randomNumber가 이동 유무를 판단하도록 구현하는 것은 어떨까?